Sources/MapboxMaps/Foundation/BaseMapView.swift

```diff
@@ -483,6 +389,48 @@ open class BaseMapView: UIView, MapClient, MBMMetalViewProvider, CameraViewDeleg
     }
 }
 
+extension BaseMapView: DelegatingObserverDelegate {
+    /// Notify correct handler
+    internal func notify(for event: MapboxCoreMaps.Event) {
+        let handlers = eventHandlers[event.type]
+        handlers?.forEach { (handler) in
+            handler(event)
+        }
+    }
+}
+
+extension BaseMapView: DelegatingMapClientDelegate {
+    internal func scheduleRepaint() {
+        needsDisplayRefresh = true
+    }
+
+    internal func scheduleTask(forTask task: @escaping Task) {
+        fatalError("scheduleTask is not supported")
+    }
+
+    internal func getMetalView(for metalDevice: MTLDevice?) -> MTKView? {
+        let metalView = MTKView(frame: frame, device: metalDevice)
+        displayCallback = {
+            metalView.setNeedsDisplay()
+        }
+
+        metalView.autoresizingMask = [.flexibleHeight, .flexibleWidth]
+        metalView.autoResizeDrawable = true
+        metalView.contentScaleFactor = UIScreen.main.scale
+        metalView.contentMode = .center
+        metalView.isOpaque = isOpaque
+        metalView.layer.isOpaque = isOpaque
+        metalView.isPaused = true
+        metalView.enableSetNeedsDisplay = true
+        metalView.presentsWithTransaction = true
+
+        insertSubview(metalView, at: 0)
+        self.metalView = metalView
+
+        return metalView
+    }
+}
```

Comment:
similar to above, Can we move this to a new file called BaseMapView+DelefatingMapClientDelegate